runtime-testsuite/test/org/antlr/v4/test/runtime/ExtraTests.java

```diff
@@ -0,0 +1,53 @@
+package org.antlr.v4.test.runtime;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExtraTests {
+	public static RuntimeTestDescriptor[] getRuntimeTestDescriptors(String targetName) {
```

Comment:
These tests are useful. We can write here test for generated and very large grammars. For instance, [LargeLexer.txt](https://github.com/antlr/antlr4/blob/master/runtime-testsuite/resources/org/antlr/v4/test/runtime/descriptors/LexerExec/LargeLexer.txt) can be moved here.

Reply:
Why not include LargeLexer etc... in main section?

Reply:
Can you remind me what the purpose is for a new file ExtraTests.java ?

Reply:
Is it only way to distinguish \r\n from \n tests?

Reply:
> Why not include LargeLexer etc... in main section?

Because it contains a lot of duplicated lines. It can be replaced by a short and clear test with several lines (also, it's possible to vary the number of tokens and other parameters).

> Is it only way to distinguish \r\n from \n tests?

For now yes. But I'm working on https://github.com/antlr/antlr4/issues/1863 and other related bugs that also require big tests that ideally should be generated.

Reply:
LargeLexer can be rewritten by the following way: https://github.com/antlr/antlr4/pull/3493/files#diff-aa2cc9084de70a064d9035752f274ac7e6551b512db05f5665cc64d92bdd0d2dR57-R83

Reply:
Hmm.. I think the whole point of the descriptor files is to avoid building tests in Java but it seems like we are now creating tests in Java that manually create descriptor objects.

Reply:
Certainly I can see that creating large test grammars in line in the descriptor files is weird. I toyed with the idea of creating a special grammar that looked like <filename.g4> instead of the actual text.  Seems like that would solve our problem right?

Reply:
> Hmm.. I think the whole point of the descriptor files is to avoid building tests in Java but it seems like we are now creating tests in Java that manually create descriptor objects.

All rules have exceptions. If it's convenient, a handwritten test can be used.

> I toyed with the idea of creating a special grammar that looked like <filename.g4> instead of the actual text. Seems like that would solve our problem right? This would also work for input/output files, many of which would have different line endings

I'm failed with saving a file with a certain line ending instead of a system line ending, git does not allow it. That's why I restored your code that normalizes line endings (`\n`). Also, I don't see a big difference between the descriptor file with inlined big grammar and the descriptor file with ref to a file with big grammar. And an external tool is required for generating such big grammars (sometimes with big input and output). It looks more complicated.

Reply:
I'll try to think about this today.  One first thought. We don't need the regex anymore if we use \n. split works great and is much easier to understand.

Reply:
Ok, I'll remove regex. Definitely, now it does not make sense since we have other tests for line endings checking.

Reply:
Done. Please don't squash my commits since they are logically different.

Also, in the future, we can use use `!fixup` commits for easier changes review. Before the final PR merge, all `!fixup` are being merged to original commits for clear history. I mean `!fixup Change something` is being merged to `Change something`.

Reply:
hi. the fixup thing might be hard for me to remember and complicate my life ;)

Reply:
Ok. I just always use interactive rebase in IDEA and never have troubles :)